lib/IRGen/IRGenModule.cpp

```diff
@@ -980,7 +983,7 @@ void IRGenModule::emitAutolinkInfo() {
                                  llvm::GlobalValue::CommonLinkage,
                                  llvm::Constant::getNullValue(Int1Ty),
                                  buf.str());
-    if (Triple.isOSBinFormatCOFF())
+    if (Triple.isOSBinFormatCOFF() && !Triple.isOSCygMing())
```

Comment:
I'm adding `IRGenModule::useDllStorage()`, many lines will be replaced but not all of them.